sklearn/utils/estimator_checks.py

```diff
@@ -314,7 +336,7 @@ def set_testing_parameters(estimator):
         # of components of the random matrix projection will be probably
         # greater than the number of features.
         # So we impose a smaller number (avoid "auto" mode)
-        estimator.set_params(n_components=1)
+        estimator.set_params(n_components=2)
```

Comment:
`set_testing_params` becomes problematic now. We will potentially hide bugs by overriding explicit tests of parameter settings. We should only override these if they are currently at their default values.

Thanks for incidentally changing something in this function so that I was reminded that it was an issue!